test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -42,52 +42,85 @@ timestamp_manager::get_timestamp_manager()
+            has_oldest = false;
     }
 
-    /* The oldest and stable timestamps must always satisfy the condition that oldest <= stable. */
-    if (new_oldest_ts > new_stable_ts) {
-        std::cout << "set_timestamp: oldest timestamp " << new_oldest_ts
-                  << " must not be later than stable timestamp " << new_stable_ts << "."
-                  << std::endl;
-        return 1;
+    /* If config has stable timestamp */
+    if (has_stable) {
+        /* Cannot proceed any further, if stable timestamp value <= 0 */
+        if ((int64_t)new_stable_ts <= 0)
+            return (false);
+        /* It is a no-op to set the new stable timestamps behind the current stable timestamp. */
+        if (new_stable_ts <= conn->get_stable_ts())
+            has_stable = false;
     }
 
-    return 0;
-}
+    /* No need to validate timestamps if stable or/and oldest were behind the global values. */
+    if (!has_oldest && !has_stable)
+        return (true);
 
-int
-timestamp_manager::validate_stable_ts(uint64_t new_stable_ts, uint64_t new_oldest_ts)
-{
-    connection_simulator *conn = &connection_simulator::get_connection();
+    /* No need to validate timestamps if there is no new and no current oldest timestamp. */
+    if (!has_oldest && conn->get_oldest_ts() == 0)
+        return (true);
 
-    /* Stable timestamp can't move backward. */
-    if (new_stable_ts <= conn->get_stable_ts()) {
-        std::cout << "Stable timestamp cannot move backwards." << std::endl;
-        return 1;
-    }
+    /* No need to validate timestamps if there is no new and no current stable timestamp. */
+    if (!has_stable && conn->get_stable_ts() == 0)
+        return (true);
 
-    /* The oldest and stable timestamps must always satisfy the condition that oldest <= stable. */
-    if (new_oldest_ts > new_stable_ts) {
-        std::cout << "set_timestamp: oldest timestamp " << conn->get_oldest_ts()
-                  << " must not be later than stable timestamp " << new_stable_ts << "."
-                  << std::endl;
-        return 1;
-    }
+    /*
+     * If the oldest timestamp was not passed in the config or was behind the current oldest
+     * timestamp, modify the new_oldest_ts to the current oldest timestamp.
+     */
+    if ((!has_oldest && conn->get_oldest_ts() != 0))
+        new_oldest_ts = conn->get_oldest_ts();
+
+    /*
+     * If the stable timestamp was not passed in the config or was behind the current stable
+     * timestamp, modify the new_stable_ts to the current stable timestamp
+     */
+    if ((!has_stable && conn->get_stable_ts() != 0))
+        new_stable_ts = conn->get_stable_ts();
 
-    return 0;
+    /* Validate if stable is greater than the oldest timestamp. */
+    if (new_oldest_ts > new_stable_ts)
+        return (false);
+
+    return (true);
 }
 
-int
-timestamp_manager::validate_durable_ts(uint64_t new_durable_ts)
+bool
+timestamp_manager::validate_durable_ts(uint64_t &new_durable_ts, bool &has_durable)
```

Comment:
You don't need to pass the `int` or `bool` by reference in this case.

Reply:
Good catch, thanks.

Reply:
Updated in [14c57d6](https://github.com/wiredtiger/wiredtiger/pull/8116/commits/14c57d641529f3f5bdf72e80b41785ec5f348085)